src/DbPostgreSQLAdapter/TableRecord.cpp

```diff
@@ -0,0 +1,171 @@
+#include "stdafx.h"
+#include "TableRecord.h"
+
+#include "Field.h"
+#include "FieldValue.h"
+
+#include "DbAdapterInterface/ITableRecordSet.h"
+
+#include "libpq-fe.h"
+
+namespace {
+	std::chrono::system_clock::time_point getDateTimeFromISOString(const std::string& ISODateTime)
+	{
+		if (!ISODateTime.empty())
+		{
+			std::chrono::system_clock::time_point result;
+			std::istringstream{ ISODateTime } >> std::chrono::parse("%F %T%z", result);
+			return result;
+		}
+
+		return std::chrono::system_clock::time_point{};
+	}
+}
+
+namespace systelab { namespace db { namespace postgresql {
+
+	TableRecord::TableRecord(ITableRecordSet& recordSet, const PGresult* statementResult, const int rowIndex)
```

Comment:
This is a problem comming from the interface in the first place, **TableRecord** (and related classes) should be a derived class from **Record** (and derived classes) because they are almost the same except that ones are tied to a table while the others not.
We have a lot code duplicated because of not adding proper hierarchy between them.

Reply:
Yep, probably we should make a refactor in a future story..